### ISSUE TITLE
ERROR: Python.exe is not found

### ISSUE THREAD

**Issue by FrenzyDemon** [AHAAAAAAA/PokemonGo-Map#3023]
## Expected Behavior

I used Location Generator to create beehive.bat, but it won't run properly. Does anyone know how I can correct this pathing issue? Python.exe is located in my default folder, so I'm not sure why.

## Current Behavior

![untitled](https://cloud.githubusercontent.com/assets/20687581/17301753/ac0a364c-57e6-11e6-8268-7c068ac30cf9.png)

## Your Environment

* Version used: Dev
* Environment name and version (e.g. Python 2.7): Python 2.7.12
* Operating System and version (desktop or mobile): Windows 10 Desktop

**Comment by freakjam** [comment]
U need to set the python path variable. Please check the wiki for instructions.

**Comment by FrenzyDemon** [comment]
I followed the instructions but I still receive same error. Can you please look at these 2 Screenshots and confirm I set Path Variable correctly?
![untitled](https://cloud.githubusercontent.com/assets/20687581/17302213/2ad9cfd0-57e9-11e6-800e-d247a49fc6ca.png)
![untitled2](https://cloud.githubusercontent.com/assets/20687581/17302214/2c7c3198-57e9-11e6-9b00-aa8a91f04826.png)

**Comment by HibikiFox** [comment]
okay so... what you are doing is trying to run the multi hex or a "beeHive"?

**Comment by FrenzyDemon** [comment]
I'm sorry, I don't understand what the difference is. My knowledge here is very limited. When I did this previously, there were instructions to create your own beehive.bat file by using a certain Google Spreadsheet.

Now, the Location Generator apparently creates the beehive.bat file for you also. All I know is that beehive.bat is the magic button that made my map work last time, and now it's not =(.

I'm sorry, I know it's frustrating dealing with computer ignorant people, but I've put so many hours into this and I'm trying my best.

**Comment by HibikiFox** [comment]
you setup the file wrong for the config for  the location generator, that is why you are seeing "the current Directory is invalid"

ERROR: Python.exe is not found

This is normal and is a result of the .bat file first killing any python tasks and then restarting the beehive. I believe this is for memory leak issues, not sure though.

So really, your error isn't `ERROR: Python.exe is not found` it is `The current directory is invalid`

Check your config file for location generator and try again

**Comment by FrenzyDemon** [comment]
I setup which file wrong? The config.ini or locationgenerator.py?

**Comment by HibikiFox** [comment]
sorry, the arguments used when running locationgenerator.py, you setup the file path wrong there... you have to set the windows file path to be where you unzipped the files from the dev download, or git pull if you do that, if you literally copy and pasted the example command from the instructions, it will not work unless you change the path, or unless you unzip the files into  `c:\pokemongo-map`

**Comment by FrenzyDemon** [comment]
python location_generator.py -lat 35.921919 -lon -84.0467687 -st 3 -lp 3  --windows --installdir="C:\Users\Tarek\Downloads\PokemonGo-Map-develop\PokemonGo-Map-develop"

That is my exact input. I don't know what I could possibly be doing wrong. I just uninstalled/deleted literally everything, and started from scratch. I am still getting the same result. I appreciate you taking the time to help

**Comment by HibikiFox** [comment]
run the command again and screen shot the error, there might be something wrong inside the .bat file

**Comment by Caspiel86** [comment]
I am running into a similar issue. I run the command below from i assume the correct CMD window according to the wiki. Account.txt is set up correctly. i get the batch created. Then i have no clue according the wiki i should get coordinates and i should use a google sheet. Then I am lost. I run the batch file and get the same error as FrenzyDemon. I also have the correct environment path for Python. Any help would be appreciated of a link to the correct instructions for making the beehive work correctly


D:\Downloads\PokemonGo-Map-develop\PokemonGo-Map-develop\Tools\Hex-Beehive-Generator>C:\Python27\python location_generator.py -lat 41.1722479 -lon -73.1997443 -st 3 -lp 3 --accounts D:\Downloads\ptc\accounts.txt --windows --installdir="D:\Downloads\PokemonGo-Map-develop\PokemonGo-Map-develop" --auth ptc
Reading usernames/passwords from D:\Downloads\ptc\accounts.txt
Generating beehive script to ../../beehive.bat

**Comment by HibikiFox** [comment]
Screen shot the command window when running the beehive.bat and post it here, 

and again `ERROR: Python.exe is not found` is normal for the first run, it is trying to reset the python process if it was running

**Comment by Caspiel86** [comment]
I figured it on on my end There were still # on the config file for the SQL server information. Once deleted and saved worked like a charm

**Comment by Caspiel86** [comment]
and i just locked out 15 accounts lol. I have to recheck my accounts.txt . i have my accounts set up like below. do they require a space before and after the comma.

user,pass
user,pass
user,pass

**Comment by FrenzyDemon** [comment]
HibikiFox, it doesn't say invalid directory anymore. Everytime "Moveable" comes up on the screen, it opens a _secondary_ window and then immediately closes out of it. After going through all "Moveables", the _original_ command window closes as well.

**Comment by HibikiFox** [comment]
~~HURRAY! you are past the easy part!~~

just kidding kinda....

So now you are missing a new dependency, What you aren't seeing is that window that opens and closes is your webserver portion, and it is demanding run something first.

You need to install NodeJS (small blurb about this in the [wiki](https://github.com/AHAAAAAAA/PokemonGo-Map/wiki) under the Windows install section) Once that is done, on either a cmd line or from a GIT bash, you need to run `NPM install`

That will run the grunt build installer that is crashing your initial window/webserver portion.

**Comment by FrenzyDemon** [comment]
I have already done NodeJS, NPM Install and Grunt Build. They all worked successfully as far as I could tell.

I will attach my beehive.bat... but I don't see any personal information in it.
[beehive.txt](https://github.com/AHAAAAAAA/PokemonGo-Map/files/394935/beehive.txt)

**Comment by HibikiFox** [comment]
Have you confirgured the Config.ini file?

**Comment by FrenzyDemon** [comment]
Yes, I added Auth, Username, Password, Location and Gmaps key. and I removed the # infront of each of those lines and I left them infront of the rest of them. Maybe I'm messing up the syntax? Are there supposed to be quotes or anything like that?

**Comment by FrenzyDemon** [comment]
I was able to take a screen shot of one of the windows before it "immediately" closed. 
![untitled](https://cloud.githubusercontent.com/assets/20687581/17312351/077f06cc-5820-11e6-9295-e601ade14e24.png)

Does that provide any leads?

**Comment by HibikiFox** [comment]
no... if you have your GMAPS api key put into the config it should work...

**Comment by FrenzyDemon** [comment]
Instead of using Location Generator to create my beehive.bat, could I just follow the old instructions and do it by copy/pasting from the Google Spreadsheet? Would that still work?

**Comment by FrenzyDemon** [comment]
It's working! I just ran "Easy Setup" and gave it my Gmaps key and my beehive.bat ran flawlessly.
I ran Easy Setup at least a dozen times and gave it my Gmaps key everytime. I don't understand why it decided to work just now.

@HibikiFox Thank you so much for all your help. You are an incredibly patient, helpful and wonderful person. The world needs more people like you. Thank you for bearing with me.
